lib/benchee/profile.ex

```diff
@@ -0,0 +1,98 @@
+defmodule Benchee.Profile do
+  alias Benchee.Output.ProfilePrinter, as: Printer
+  alias Benchee.Suite
+
+  defmodule Benchee.UnknownProfilerError do
+    defexception message: "error"
+  end
+
+  @moduledoc """
+  Profiles each scenario after benchmarking them if the `profile_after` option is either set to:
+    * `true`,
+    * a valid `profiler`,
+    * a tuple of a valid `profiler` and a list of options to pass to it, e.g., `{:fprof, [sort: :own]}`.
+
+  The profiler that will be used is either the one set by the `profiler_after` option or, if set to `true`,
+  the default one (`:cprof`). It accepts however the following profilers:
+    * `:cprof` will profile with `Mix.Task.Profile.Cprof`. It provides information related to the
+    number of function calls.
+    * `:eprof` will profile with `Mix.Task.Profile.Eprof`. It provides information related to the
+    time spent on each function in regard to the total execution time.
+    * `:fprof` will profile with `Mix.Task.Profile.Fprof`. It provides information related to the
+    time spent on each function, both the *total* time spent on it and the time spent on it,
+    *excluding* the time of called functions.
+  """
+
+  @doc """
+  Runs for each scenario found in the suite the `profile/2` function from the given profiler.
+  """
+  @default_profiler :cprof
+  @spec profile(Suite.t()) :: Suite.t()
+  def profile(suite, printer \\ Printer)
+  def profile(suite = %{configuration: %{profile_after: false}}, _printer), do: suite
+
+  def profile(suite = %{configuration: %{profile_after: true}}, printer) do
+    config = %{suite.configuration | profile_after: {@default_profiler, []}}
+
+    %{suite | configuration: config}
+    |> do_profile(printer)
```

Comment:
this is a mnior thing that we can also tackle afterwards:

Adding it to the config and then calling `do_profile` with it doesn't feel quite right although it works.

I think it might be nicer if `do_profile` already accepted directly what it needed. So instead of mangling this we could call something like:

```elixir
do_profile(scenarios, {@default_profiler, []}, printer)
```

Seems cleaner for me and a nicer/more reduced interface for `do_profile`. This way `profile` pulls the arguments out the right way for `do_profile` basically :)

Reply:
Yeah, I can see it being more clean and less hacky. Changed it! :)